internal/dispatch/graph/reachableresources_test.go

```diff
@@ -603,3 +609,109 @@ func TestCaveatedReachableResources(t *testing.T) {
 		})
 	}
 }
+
+func TestReachableResourcesWithConsistencyLimitOf1(t *testing.T) {
```

Comment:
Does this test reproduce the "possible deadlock" as well? The PR body does not describe how the deadlock occurs, so I assume it has to do with unschedulable dispatch operations. How did that happen with the errgroup?

Reply:
Yes, it was the first test I wrote to demonstrate it